fastapi/_compat.py

```diff
@@ -478,7 +489,11 @@ def get_definitions(
         )
 
     def is_scalar_field(field: ModelField) -> bool:
-        return is_pv1_scalar_field(field)
+        from fastapi import params
+
+        return is_pv1_scalar_field(field) and not isinstance(
+            field.field_info, params.Body
```

Comment:
This additional `params.Body` is same as in PydanticV2 `def is_scalar_field` side